x/nft-transfer/ibc_module.go

```diff
@@ -0,0 +1,291 @@
+package nfttransfer
+
+import (
+	"fmt"
+	"math"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
+
+	channeltypes "github.com/cosmos/ibc-go/v5/modules/core/04-channel/types"
+	porttypes "github.com/cosmos/ibc-go/v5/modules/core/05-port/types"
+	host "github.com/cosmos/ibc-go/v5/modules/core/24-host"
+	ibcexported "github.com/cosmos/ibc-go/v5/modules/core/exported"
+	"github.com/crypto-org-chain/chain-main/v4/x/nft-transfer/keeper"
+	"github.com/crypto-org-chain/chain-main/v4/x/nft-transfer/types"
+)
+
+var _ porttypes.IBCModule = IBCModule{}
+
+// IBCModule implements the ICS26 interface for transfer given the transfer keeper.
+type IBCModule struct {
+	keeper keeper.Keeper
+}
+
+// NewIBCModule creates a new IBCModule given the keeper
+func NewIBCModule(k keeper.Keeper) IBCModule {
+	return IBCModule{
+		keeper: k,
+	}
+}
+
+// ValidateTransferChannelParams does validation of a newly created nft-transfer channel. A nft-transfer
+// channel must be UNORDERED, use the correct port (by default 'nft-transfer'), and use the current
```

Comment:
in: https://github.com/cosmos/ibc/tree/main/spec/app/ics-721-nft-transfer#port--channel-setup
the port is called just "nft"?

Reply:
Changed

Reply:
nit: comment still says `nft-transfer`

Reply:
Changed